facebook_connector.py

```diff
@@ -0,0 +1,110 @@
+"""
+
+Author: Nigel Schuster <nigel schusters at gmail dot com>
+
+This module provides a simple abstraction for accessing the Facebook chat data
+
+"""
+
+from __future__ import print_function
+
+from bs4 import BeautifulSoup
+from fbchat_archive_parser.parser import MessageHtmlParser
+import os
+import pandas as pd
+import re
+import warnings
+
+_messages_file = None
+_fb_id_pattern = re.compile(r"(\d+)@facebook\.com")
+
+
+def initialize(dump_directory="."):
+    """
+        Initializes the access to the Facebook messages file by asserting its location
+
+        Args:
+            dump_directory: path to the directory that contains messages.htm file
+    """
+    global _messages_file
+    fb_message_filename = "messages.htm"
+    _messages_file = os.path.join(dump_directory, fb_message_filename)
+    if not os.path.isfile(_messages_file):
+        print("Please specify the location of the file in the function call")
+        _messages_file = None
+
+
+def resolve_user_id(user_id):
+    """
+    Tries to map the identifier facebook provides to the name of the user
+
+    Args:
+        user_id: identifier string that is provided in the messages file
+
+    Returns:
+        The name of the user if it is able to find it, otherwise the input
+    """
+    # TODO (neitsch): This method is not implemented yet
+    #                 Use the user_id (format: 12345678@facebook.com) to get the
+    #                 actual name of the user
+    return user_id
+
+
+def get_cleaned_fully_merged_messages(strip_html_content=True,
+                                      resolve_fb_id=False):
+    """
+    Parses the messages file to create dataframes that contain the messages and their senders
+
+    Args:
+        strip_html_content: The messages.htm file might contain some html tags in messages; this option will remove all html markup
+        resolve_fb_id: The messages.htm file doesn't always print Facebook names, but sometimes ids instead; this will atempt to resolve the issue,
+		but requires a web request per id and is not guaranteed to work
+
+    Returns:
+        a dataframe that contains all messages with info about their senders
+    """
+    if _messages_file is None:
+        print("Please initialize the module first.")
+        return
+    chats = MessageHtmlParser(path=_messages_file).parse()
+    me = chats.user
+    addresses = set()
+    messages = []
+    # Suppressing warning on BS4 that shows up when trying to parse non HTML content
+    # Some messages might contain HTML tags others might not and we don't want to display
+    # these warnings to the user
```

Comment:
See my comment in the "conversation section."  Makes sense to keep this, but the comment could use some clarification